src/build.rs

```diff
@@ -118,8 +117,7 @@ impl BuildSystem {
         let staging_dist = self.cfg.staging_dist.clone();
 
         let mut entries = fs::read_dir(&staging_dist).await.context("error reading staging dist dir")?;
-        while let Some(entry) = entries.next().await {
-            let entry = entry.context("error reading contents of staging dist dir")?;
```

Comment:
This context is kinda important, or it'll end up skipping this while loop without an error. Either that or we need to allow `next_entry` to be short-circuted with a `?`